### ISSUE TITLE
'@' in compilerArgs doesn't work on Linux/Mac with 1.10.5 (Intellisense breaks)

### ISSUE THREAD

**Issue by maxgerhardt** [microsoft/vscode-cpptools#9434]
### Environment

- OS and version: Ubuntu 20.04.4 LTS
- VS Code: 1.68.0
- C/C++ extension: 1.10.5
- OS and version of remote machine (if applicable): -/-
- GDB / LLDB version: not applicable.

### Bug Summary and Steps to Reproduce

Bug Summary:

When using PlatformIO with a Raspberry Pi Pico + Arduino project on Unix-like operating systems (Linux + Mac), Intellisense fails to find all include paths. Reverting the C/C++ by Microsoft extension to 1.9.8 solves the problem. The current version 1.10.5 that you have released 3 days ago breaks it.

Steps to reproduce:
1. Install the PlatformIO extension
2. Create a new project, choose "Raspberry Pi Pico" as board and "Arduino" framework
3. Let the default `src/main.cpp` code be.
4. Use the [project task](https://docs.platformio.org/en/latest/integration/ide/vscode.html#project-tasks) "Build", should be successfull
5. Observe intellisense error on `#include <Arduino.h>` line

### Debugger Configurations

Not relevant for Intellisense. Only `launch.json` is auto-generated by PlatformIO with 
```json
// AUTOMATICALLY GENERATED FILE. PLEASE DO NOT MODIFY IT MANUALLY
//
// PIO Unified Debugger
//
// Documentation: https://docs.platformio.org/page/plus/debugging.html
// Configuration: https://docs.platformio.org/page/projectconf/section_env_debug.html

{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "platformio-debug",
            "request": "launch",
            "name": "PIO Debug",
            "executable": "/home/max/p_test/.pio/build/pico/firmware.elf",
            "projectEnvName": "pico",
            "toolchainBinDir": "/home/max/.platformio/packages/toolchain-gccarmnoneeabi/bin",
            "internalConsoleOptions": "openOnSessionStart",
            "svdPath": "/home/max/.platformio/platforms/raspberrypi/misc/svd/rp2040.svd",
            "preLaunchTask": {
                "type": "PlatformIO",
                "task": "Pre-Debug"
            }
        },
        {
            "type": "platformio-debug",
            "request": "launch",
            "name": "PIO Debug (skip Pre-Debug)",
            "executable": "/home/max/p_test/.pio/build/pico/firmware.elf",
            "projectEnvName": "pico",
            "toolchainBinDir": "/home/max/.platformio/packages/toolchain-gccarmnoneeabi/bin",
            "internalConsoleOptions": "openOnSessionStart",
            "svdPath": "/home/max/.platformio/platforms/raspberrypi/misc/svd/rp2040.svd"
        },
        {
            "type": "platformio-debug",
            "request": "launch",
            "name": "PIO Debug (without uploading)",
            "executable": "/home/max/p_test/.pio/build/pico/firmware.elf",
            "projectEnvName": "pico",
            "toolchainBinDir": "/home/max/.platformio/packages/toolchain-gccarmnoneeabi/bin",
            "internalConsoleOptions": "openOnSessionStart",
            "svdPath": "/home/max/.platformio/platforms/raspberrypi/misc/svd/rp2040.svd",
            "loadMode": "manual"
        }
    ]
}
```

### Debugger Logs

```shell
None relevant.
```

### Other Extensions

PlatformIO: 2.4.3

### Additional Information

Project with C/C++ version 1.10.5:

![grafik](https://user-images.githubusercontent.com/26485477/173248646-c51cef13-612e-4211-bb2a-6222623d1b32.png)

**EXACT SAME** project with C/C++ version 1.9.8: 

![grafik](https://user-images.githubusercontent.com/26485477/173248673-2527238d-6d2a-4ce4-866b-e34db884016b.png)

Reference `.vscode/c_cpp_properties.json`
```json
//
// !!! WARNING !!! AUTO-GENERATED FILE!
// PLEASE DO NOT MODIFY IT AND USE "platformio.ini":
// https://docs.platformio.org/page/projectconf/section_env_build.html#build-flags
//
{
    "configurations": [
        {
            "name": "PlatformIO",
            "includePath": [
                "/home/max/p_test/include",
                "/home/max/p_test/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino",
                "/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino/api/deprecated",
                "/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino/api/deprecated-avr-comp",
                "/home/max/.platformio/packages/framework-arduino-mbed/variants/RASPBERRY_PI_PICO",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Camera/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Ethernet/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/GC2145",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/GPS/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/GSM/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Himax_HM01B0",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/KernelDebug/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/MCUboot/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/MLC/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/MRI/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Nano33BLE_System/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Nicla_System/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/PDM/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_Audio/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_SDCARD/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_SDRAM/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_Video/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_lvgl/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/RPC/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SE05X/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SFU/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SPI",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/STM32H747_System/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Scheduler/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SocketWrapper/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/ThreadDebug/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBAudio",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBHID/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBHOST/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBMSD/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/WiFi/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Wire",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/doom/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/ea_malloc",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/mbed-memory-status",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/openamp_arduino/src",
                "/home/max/.platformio/packages/framework-arduino-mbed/libraries/rpclib/src",
                ""
            ],
            "browse": {
                "limitSymbolsToIncludedHeaders": true,
                "path": [
                    "/home/max/p_test/include",
                    "/home/max/p_test/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino",
                    "/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino/api/deprecated",
                    "/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino/api/deprecated-avr-comp",
                    "/home/max/.platformio/packages/framework-arduino-mbed/variants/RASPBERRY_PI_PICO",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Camera/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Ethernet/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/GC2145",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/GPS/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/GSM/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Himax_HM01B0",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/KernelDebug/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/MCUboot/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/MLC/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/MRI/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Nano33BLE_System/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Nicla_System/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/PDM/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_Audio/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_SDCARD/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_SDRAM/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_Video/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Portenta_lvgl/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/RPC/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SE05X/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SFU/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SPI",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/STM32H747_System/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Scheduler/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/SocketWrapper/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/ThreadDebug/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBAudio",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBHID/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBHOST/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/USBMSD/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/WiFi/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/Wire",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/doom/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/ea_malloc",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/mbed-memory-status",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/openamp_arduino/src",
                    "/home/max/.platformio/packages/framework-arduino-mbed/libraries/rpclib/src",
                    ""
                ]
            },
            "defines": [
                "PLATFORMIO=60003",
                "ARDUINO_RASPBERRY_PI_PICO",
                "ARDUINO_ARCH_RP2040",
                "ARM_MATH_CM0PLUS",
                "__CMSIS_RTOS",
                "COMPONENT_FLASHIAP=1",
                "__CORTEX_M0PLUS",
                "DEVICE_ANALOGIN=1",
                "DEVICE_FLASH=1",
                "DEVICE_I2C=1",
                "DEVICE_I2CSLAVE=1",
                "DEVICE_INTERRUPTIN=1",
                "DEVICE_PORT_IN=1",
                "DEVICE_PORT_OUT=1",
                "DEVICE_PWMOUT=1",
                "DEVICE_RESET_REASON=1",
                "DEVICE_RTC=1",
                "DEVICE_SERIAL=1",
                "DEVICE_SERIAL_FC=1",
                "DEVICE_SPI=1",
                "DEVICE_USBDEVICE=1",
                "DEVICE_USTICKER=1",
                "DEVICE_WATCHDOG=1",
                "__MBED__=1",
                "MBED_BUILD_TIMESTAMP=1652255920.4792204",
                "__MBED_CMSIS_RTOS_CM",
                "MBED_MPU_CUSTOM",
                "MBEDTLS_ENTROPY_NV_SEED",
                "PICO_NO_BINARY_INFO=1",
                "PICO_ON_DEVICE=1",
                "PICO_RP2040_USB_DEVICE_ENUMERATION_FIX=1",
                "PICO_TIME_DEFAULT_ALARM_POOL_DISABLED",
                "PICO_UART_ENABLE_CRLF_SUPPORT=0",
                "TARGET_CORTEX",
                "TARGET_CORTEX_M",
                "TARGET_LIKE_CORTEX_M0",
                "TARGET_LIKE_MBED",
                "TARGET_M0P",
                "TARGET_memmap_default",
                "TARGET_NAME=RASPBERRY_PI_PICO",
                "TARGET_RASPBERRYPI",
                "TARGET_RASPBERRY_PI_PICO",
                "TARGET_RELEASE",
                "TARGET_RP2040",
                "TOOLCHAIN_GCC",
                "TOOLCHAIN_GCC_ARM",
                "MBED_NO_GLOBAL_USING_DIRECTIVE=1",
                "CORE_MAJOR=3",
                "CORE_MINOR=1",
                "CORE_PATCH=1",
                "USE_ARDUINO_PINOUT",
                "ARDUINO=10810",
                "ARDUINO_ARCH_MBED",
                ""
            ],
            "cStandard": "c11",
            "cppStandard": "c++14",
            "compilerPath": "/home/max/.platformio/packages/toolchain-gccarmnoneeabi/bin/arm-none-eabi-gcc",
            "compilerArgs": [
                "-mcpu=cortex-m0plus",
                "-mthumb",
                "-iprefix/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino",
                "@/home/max/.platformio/packages/framework-arduino-mbed/variants/RASPBERRY_PI_PICO/includes.txt",
                ""
            ]
        }
    ],
    "version": 4
}
```

Also discussed in https://github.com/platformio/builder-framework-arduino-core-mbed/issues/4

Intellisense can be made to work with the latest extension when transforming the 
```
                "-iprefix/home/max/.platformio/packages/framework-arduino-mbed/cores/arduino",
                "@/home/max/.platformio/packages/framework-arduino-mbed/variants/RASPBERRY_PI_PICO/includes.txt",
```
part into seperate entries in the `path` array.

My guess: Recent versions don't parse `-iprefix <path> @<include file>` correctly. 

Include files come from https://github.com/arduino/ArduinoCore-mbed, includes.txt from https://github.com/arduino/ArduinoCore-mbed/blob/master/variants/RASPBERRY_PI_PICO/includes.txt.

**Comment by maxgerhardt** [comment]
Note: Due to a higher number of users, this is kinda critical.

![grafik](https://user-images.githubusercontent.com/26485477/173248951-4bd861f2-76e6-49a2-a3fd-1b42a0f87815.png)

**Comment by ivankravets** [comment]
@sean-mcmanus, @Colengms, @robotdad we can't fix this issue on our side. The problem is linked with Arduino/ARM Mbed core. They have a ton of includes and the only solution to avoid the Windows CMD MAXLEN limit (toolchain crash) is to wrap includes into the txt file.

Could we ask you to revert back support for `-ipreifx/ @`?

Thanks.

**Comment by bobbrow** [comment]
Can you try `"C_Cpp.legacyCompilerArgsBehavior": true`?  The team will follow up with additional guidance if that does not work for you.

**Comment by ahmedalkabir** [comment]
the latest version doesn't get well with the Platformio as well, now I get "Unable to resolve configuration with compilerPath" although the location is valid, I downgraded vscode-cpptools to resolve the issue.

@sean-mcmanus I hope you'll resolve the issue as soon as possible, and thanks.

**Comment by sean-mcmanus** [comment]
@bobbrow legacyCompilerArgsBehavior doesn't fix the issue.

I've reproed the bug with a simple sample config -- the issue is not with -iprefix, it's with the handling of `@`. Something regressed that. I'm investigating...

**Comment by sean-mcmanus** [comment]
Also, the bug only repros on Linux (probably Mac too) and not Windows. As far as we know, the `@` response files are only required on Windows, but we should still support their usage on Linux/Mac.

**Comment by maxgerhardt** [comment]
>Can you try "C_Cpp.legacyCompilerArgsBehavior": true? The team will follow up with additional guidance if that does not work for you.

I experience the same the same issue with the `"C_Cpp.legacyCompilerArgsBehavior": true` option, so that's in line with @sean-mcmanus.

![grafik](https://user-images.githubusercontent.com/26485477/173462917-8b0b6f73-3aff-4b52-b5bd-3cd5c1ffb06c.png)

**Comment by bobbrow** [comment]
Thanks for confirming.  @sean-mcmanus has proposed a fix and we are currently reviewing it.

**Comment by sean-mcmanus** [comment]
We should be able to release the fix tomorrow (end of day) in a 1.10.6 unless we find some reason to delay it.

**Comment by ivankravets** [comment]
Thanks, @sean-mcmanus and @bobbrow, for the quick response and fix! 🙏

**Comment by VarunArora14** [comment]
In windows also there is a problem in 1.10.5 where it says "Unable to spawn Intellisense" in output tab and in Problem tab, keeps searching for Include path. I got back to version 1.98 but 1.10.5 is just not working on windows too.

**Comment by bobbrow** [comment]
@sean-mcmanus did you see @VarunArora14's problem at all on Windows?

**Comment by sean-mcmanus** [comment]
@bobbrow @VarunArora14 That sounds like a completely different issue: https://github.com/microsoft/vscode-cpptools/issues/9431 . We have not yet gotten a repro or a diagnosis as to what is causing it, but it seems like a major regression that is being hit a lot of users.

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/v1.10.6

**Comment by sean-mcmanus** [comment]
If anyone was hitting an IntelliSense regression failure on Windows, that should be fixed with 1.10.7.
